### PR TITLE
create basic authentication, create layout

### PR DESCRIPTION
**Pull request by sondh5**

### CODE REVIEW

**Comment by hongsondhs** [comment]
inprogess

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 class ApplicationController < ActionController::Base
+	http_basic_authenticate_with name: ENV["BASIC_AUTHEN_USERNAME"], password: ENV["BASIC_AUTHEN_PASSWORD"]
```
Comment (lucnv) [inline app/controllers/application_controller.rb]: Indent

Comment (sondh5) [reply app/controllers/application_controller.rb]: ở local nó bằng mấy cái kia mà @@~

File: app/controllers/static_pages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      render template: "static_pages/#{params[:page]}"
+    else
+      render_404
+    end
+  end
+
+  private
+  def valid_page?
+    File.exist?(Pathname.new(Rails.root +
+      "app/views/static_pages/#{params[:page]}.html.erb"))
```
Comment (lucnv) [inline app/controllers/static_pages_controller.rb]: File.exist? Pathname.new("#{Rails.root}/app/views/static_pages/#{params[:page]}.html.erb"))

**Review by cuongnc0211 — APPROVED** [review]

File: app/views/layouts/_footer.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<footer class="footer">
+  <small>
+    <p>© <%= Date.today.year %> <%= t ".rooms_crawler" %></p>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to t(".about"), "/about" %></li>
```
Comment (lucnv) [inline app/views/layouts/_footer.html.erb:7]: wrong url

Comment (sondh5) [reply app/views/layouts/_footer.html.erb:7]: check routes.
chạy ngon nhé

**Comment by lucnv** [comment]
ok

**Pull request merged**